tla-types/src/main/scala/at/forsyte/apalache/tla/typecheck/etc/ToEtcExpr.scala

```diff
@@ -0,0 +1,594 @@
+package at.forsyte.apalache.tla.typecheck.etc
+
+import at.forsyte.apalache.tla.lir._
+import at.forsyte.apalache.tla.lir.oper._
+import at.forsyte.apalache.tla.lir.values._
+import at.forsyte.apalache.tla.typecheck._
+import at.forsyte.apalache.tla.typecheck.parser.{DefaultType1Parser, Type1ParseError}
+
+/**
+  * ToSTCExpr takes a TLA+ expression and produces an STCExpr.
+  * The most interesting part of this translation is dealing with the built-in operators.
+  * This translator is an extension of the ideas that appear in SignatureGenerator by Jure Kukovec.
+  *
+  * @see at.forsyte.apalache.tla.types.SignatureGenerator
+  *
+  * @author Igor Konnov
+  */
+class ToEtcExpr extends EtcBuilder {
+  private val type1Parser = DefaultType1Parser
+
+  // TODO: add support for recursive functions
+
+  /**
+    * Translate an operator declaration.
+    *
+    * @param decl an operator declaration
+    * @param inScopeEx an expression to chain in the bottom of let-definition, it should have been translated
+    * @return the translated let-in expression with inScopeEx attached
+    */
+  def apply(decl: TlaOperDecl, inScopeEx: EtcExpr): EtcExpr = {
+    val paramsAndDoms = formalParamsToTypeVars(0, decl.formalParams).map {
+      case (paramName, paramType) =>
+        (paramName, mkConst(BlameRef(decl.body.ID), SetT1(paramType)) )
+    }
+
+    if (decl.name.startsWith("TypeAssumptions")) {
+      // the special form for type annotations over state variables and constants
+      parseTypeAssumptions(decl, inScopeEx)
```

Comment:
Breaking this out into a helper function makes it easy to separate the control flow from the actions dispatched on the different branches of the match statement, and the function name helps self document the intent. I like it! 

I think you could remove the `case 1: ....` and `case 2: ...` comments below if you also refactored those out into helper functions. That would make this section of code more consistent (why is `parseTypeAssumptions` a helper function but there's no helper for `parseTypedOperBody`?) and imo help with readability in general.